### PR TITLE
Allow clicking on percentile charts

### PR DESCRIPTION
**Pull request by jdhollen**
I'll be using this on some charts for build times.

There isn't a concrete click handler for lines that provides the "closest" point, so I had to improvise a bit: this just uses the last date / x-axis value that we showed a tooltip for.  Since the percentile chart *always* shows a tooltip on hover, this seems alright to me--the user's click action will match the date that recharts thought they were hovering on.

### CODE REVIEW

**Comment by brentleyjones** [comment]
Does this still work for mobile (thinking about how hover works differently there)?

File: enterprise/app/trends/percentile_chart.tsx [hunk 1]
```diff
@@ -100,9 +115,11 @@ function PercentilesChartTooltip({
   extractP90,
   extractP95,
   extractP99,
+  triggerCallback,
 }: PercentilesChartTooltipProps) {
   if (!active) return null;
   let data = payload[0].payload;
+  triggerCallback(data);
```
Comment (bduffany) [inline enterprise/app/trends/percentile_chart.tsx:122]: nit: IIRC the React docs strongly recommend against having `render()` do anything that can mutate state. Would it work to put this in a class component and maybe put this in `componentDidUpdate`?

**Comment by jdhollen** [comment]
@brentleyjones this *should* at least click out to the right thing on mobile because a hover event will fire prior to the click, but obviously it busts up "tap to see the tooltip" when the user does something that the browser counts as a click.

we currently have the same problem with bar charts on mobile: tapping on a bar with a click handler will immediately take you through.

we could do something gnarly where we validate that the tooltip is (a) showing and (b) matches the current value that we think the user just clicked on (i.e., two taps to actually click through on mobile).  this is a bit of a pain to test with a local instance, though.  how bout i get this out to dev and then go from there?  this code is unused / will be flag-guarded anyway.

File: enterprise/app/trends/percentile_chart.tsx (hunk 1 above)
Comment (jdhollen) [reply enterprise/app/trends/percentile_chart.tsx]: Sure yeah done

**Pull request merged**